packages/material-ui/src/Select/Select.js

```diff
@@ -191,6 +191,8 @@ Select.propTypes = {
    * The input value. Providing an empty string will select no options.
    * This prop is required when the `native` prop is `false` (default).
    * Set to an empty string `''` if you don't want any of the available options to be selected.
+   *
+   * ⚠️Value comparison to each option is a reference equality check "===", different object types will never be equal.
```

Comment:
We only use `===` for objects. For anything else we use equality of the stringified values.

I'm also not sure `different object types` is appropriate. This sounds like it's doing value comparison. `{ id: 1 } !== { id: 1 }` even though there types are equal.

Reply:
Thank you for explaining, I will review carefully and attempt to document again